cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -61,7 +61,6 @@ func NewSimulateTransactionHandler(logger *log.Entry, ledgerEntryReader db.Ledge
 		if opSourceAccount := op.SourceAccount; opSourceAccount != nil {
 			sourceAccount = opSourceAccount.ToAccountId()
 		} else {
-			// FIXME: SourceAccount() panics, so, the user can doctor an envelope which makes the server crash
 			sourceAccount = txEnvelope.SourceAccount().ToAccountId()
```

Comment:
Functions from the `xdr` package do panic, but the XDR decoder makes sure no incorrect envelope type can be obtain after parsing the XDR.